e2e/nomostest/clusters/gke.go

```diff
@@ -108,8 +112,25 @@ func waitOperation(ctx context.Context, t testing.NTB, operation string) error {
 	return nil
 }
 
-func listAndWaitForOperations(ctx context.Context, t testing.NTB, name string) error {
-	ctxWithTimeout, cancel := context.WithTimeout(ctx, 5*time.Minute)
+type operationOptions struct {
+	timeout time.Duration
+}
+type operationOption func(opts *operationOptions)
+
+func withOperationTimeout(timeout time.Duration) operationOption {
+	return func(opts *operationOptions) {
+		opts.timeout = timeout
+	}
+}
+
+func listAndWaitForOperations(ctx context.Context, t testing.NTB, name string, opts ...operationOption) error {
+	options := &operationOptions{
```

Comment:
do we need all this operationOptions complexity? why not just pass in a context with a timeout defined by the caller?

Reply:
If you want to define a default timeout you can use ctx.Deadline() to see if the context has a timeout or not.

Reply:
done